### DISCUSSION TITLE
Improve "tap the beat!" widget

### DISCUSSION THREAD

**Discussion by MeerukYang** [ppy/osu#29581] [Ideas]
when testing a song's BPM by using "tap the beat!" widget, it would jump to the beginning of the current timing stamp. See attached video.

https://github.com/user-attachments/assets/f16aa7dc-6602-489c-b1e7-346c4fc0095b

This makes it hard to catch the beat, and you have to spend more time tapping to the beat to get an accurate BPM, which is really confusing and annoying.
My suggestion is to continue playing the song when tapping instead of jumping to the beginning of a stamp.

version: 2024.817.0

**Comment by Tenexxt** [comment]
Impractical because if the BPM changed anywhere you'd be Basically doomed to get the incorrect one

**Reply by MeerukYang** [reply]
the prerequisite of using this widget is that you having confidence that the song's BPM is constant. if BPM changes then you should never use this widget; other approachs is always better in this case.
Thus i don't think it impractical.

**Reply by Tenexxt** [reply]
There are a lot of songs that have mostly the same BPM but change at some point, which would make the widget practical if used correctly.
The widget brings you back to the timestamp so you can calculate the BPM 100% correctly in case it would change at some other point.

I'm not going to argue about this because I'm not the one deciding what will be in the game and what won't, so I'll just leave this for now.
